app/src/org/commcare/views/widgets/GregorianDateWidget.java

```diff
@@ -0,0 +1,336 @@
+
+import android.content.Context;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import org.commcare.dalvik.R;
+import org.commcare.utils.UniversalDate;
+import org.javarosa.core.model.data.DateData;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.core.model.data.InvalidData;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.joda.time.DateTime;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Created by Saumya on 5/27/2016.
+ * A widget that accepts Gregorian dates using logic and GUI that are similar to the Nepali and Ethiopian widgets
+ */
+public class GregorianDateWidget extends AbstractUniversalDateWidget {
+
+    private EditText dayTxt;
+    private AutoCompleteTextView monthTxt;
+    private EditText yearTxt;
+    private TextView dayOfWeek;
+    private Calendar selectedDate;
+    private String[] monthNames;
+    private List<String> monthList;
+    private final int MIN_YEAR = 1900;
+    private final String DAYFORMAT = "%02d";
+    private final String YEARFORMAT = "%04d";
+    private int maxYear;
+
+    protected LinearLayout gregorianView;
+
+    public GregorianDateWidget(Context context, FormEntryPrompt prompt){
+        super(context, prompt);
+        selectedDate = Calendar.getInstance();
+        maxYear = selectedDate.get(Calendar.YEAR) + 1;
+        ImageButton clearAll = (ImageButton) findViewById(R.id.clearall);
+        clearAll.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                clearAll();
+            }
+        });
+    }
+
+    @Override
+    protected void initText(){
+        dayOfWeek = (TextView) findViewById(R.id.gregdayofweek);
+        dayTxt = (EditText)findViewById(R.id.daytxtfield);
+        yearTxt = (EditText)findViewById(R.id.yeartxtfield);
+
+        MonthAdapter monthAdapter = new MonthAdapter(getContext(), monthList);
+        monthTxt = (AutoCompleteTextView) findViewById(R.id.monthtxtfield);
+        monthTxt.setAdapter(monthAdapter);
+
+        dayTxt.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {}
+
+            @Override
+            public void afterTextChanged(Editable s) {

```

Comment:
high-level discussion we had here- may want to consider making the widget's corrective behavior stronger than it currently is (i.e. immediately revert to the closest valid day/month upon clicking away if user input is invalid)